docs/source/topics/data_types.rst

```diff
@@ -4,8 +4,790 @@
+Let's begin with creating the instance by defining its unit cell:
+
+.. code-block:: ipython
+
+  In [1]: StructureData = DataFactory('structure')
+
+  In [2]: unit_cell = [[3.0, 0.0, 0.0], [0.0, 3.0, 0.0], [0.0, 0.0, 3.0]]
+
+  In [3]: structure = StructureData(cell=unit_cell)
+
+.. note::
+
+    Default units for crystal structure cell and atomic coordinates in AiiDA are Å (Ångström).
+
+Next, you can add the Li atoms to the structure using the :py:class:`~aiida.orm.nodes.data.structure.StructureData.append_atom()` method:
+
+.. code-block:: ipython
+
+  In [4]: structure.append_atom(position=(0.0, 0.0, 0.0), symbols="Li")
+
+  In [5]: structure.append_atom(position=(1.5, 1.5, 1.5), symbols="Li")
+
+You can check if the cell and sites have been set up properly by checking the ``cell`` and ``sites`` properties:
+
+.. code-block:: ipython
+
+  In [6]: structure.cell
+  Out[6]: [[3.5, 0.0, 0.0], [0.0, 3.5, 0.0], [0.0, 0.0, 3.5]]
+
+  In [7]: structure.sites
+  Out[7]: [<Site: kind name 'Li' @ 0.0,0.0,0.0>, <Site: kind name 'Li' @ 1.5,1.5,1.5>]
+
+From the :py:class:`~aiida.orm.nodes.data.structure.StructureData` node you can also obtain the formats of well-known materials science Python libraries such as `the Atomic Simulation Environment <https://wiki.fysik.dtu.dk/ase/index.html>`_ (ASE) and `pymatgen <https://pymatgen.org/index.html>`_:
+
+.. code-block:: ipython
+
+  In [8]: structure.get_ase()
+  Out[8]: Atoms(symbols='Li2', pbc=True, cell=[3.5, 3.5, 3.5], masses=...)
+
+  In [9]: structure.get_pymatgen()
+  Out[9]:
+  Structure Summary
+  Lattice
+      abc : 3.5 3.5 3.5
+  angles : 90.0 90.0 90.0
+  volume : 42.875
+        A : 3.5 0.0 0.0
+        B : 0.0 3.5 0.0
+        C : 0.0 0.0 3.5
+  PeriodicSite: Li (0.0000, 0.0000, 0.0000) [0.0000, 0.0000, 0.0000]
+  PeriodicSite: Li (1.5000, 1.5000, 1.5000) [0.4286, 0.4286, 0.4286]
+
+Exporting
+^^^^^^^^^
+
+The following export formats are available for :py:class:`~aiida.orm.nodes.data.structure.StructureData`:
+
+* ``xsf`` (format supported by e.g. XCrySDen and other visualization software;
+  supports periodic cells)
+* ``xyz`` (classical xyz format, does not typically support periodic cells (even if
+  the cell is indicated in the comment line)
+* ``cif`` (export to CIF format, without symmetry reduction, i.e. always storing the
+  structure as P1 symmetry)
+
+The node can be exported using the verdi CLI, for example:
+
+.. code-block:: console
+
+    $ verdi data structure export --format xsf <IDENTIFIER> > Li.xsf
+
+Where ``<IDENTIFIER>`` is one of the possible identifiers of the node, e.g. its PK or UUID.
+This outputs the structure in ``xsf`` format and writes it to a file.
+
+.. _topics:data_types:materials:trajectory:
+
+TrajectoryData
+--------------
+
+The :py:class:`~aiida.orm.nodes.data.array.trajectory.TrajectoryData` data type represents a sequences of StructureData objects, where the number of atomic kinds and sites does not change over time.
```

Comment:
We should emphasise here that, at variance with the StructureData, here coordinates are stored in an array and not in the DB.